### PR TITLE
Initial support for Databricks Repos

### PR DESCRIPTION
**Pull request by alexott**
This PR adds an initial support for [Databricks Repos](https://docs.databricks.com/repos.html) feature using the [Repos API](https://docs.databricks.com/dev-tools/api/latest/repos.html).  

Summary of changes:
* the new top level command was added: `repos`
* the `update` subcommand is used to checkout the repository specified via `--repos-id` switch to the branch specified via `--branch` switch
* the `get-repos-id` subcommand uses Workspaces API to find the Repos ID based on the file path.  It will be used to simplify the integration with CI/CD systems, etc., so people can write something like this instead of looking for Repos ID in UI:
```sh
REPOS_ID=$(databricks repos get-repos-id /Repos/Production/some-project)
databricks repos update --repos-id $REPOS_ID --branch releases
```

### CODE REVIEW

File: databricks_cli/repos/api.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+class ReposApi(object):
+    def __init__(self, api_client):
+        self.client = ReposService(api_client)
+        self.ws_client = WorkspaceService(api_client)
+
+    def update(self, repo_id, branch):
+        return self.client.update(repo_id, branch)
+
+    def get_repos_id(self, path):
```
Comment (anders-databricks) [inline databricks_cli/repos/api.py:36]: Can we leave this feature out for now? We're planning on extending the Repos API with additional CRUD endpoints, and they will achieve this same goal.

Comment (alexott) [reply databricks_cli/repos/api.py:36]: you mean `get_repos_id` or both?

Comment (anders-databricks) [reply databricks_cli/repos/api.py:36]: Yeah I meant just the get-repos-id subcommand!

**Comment by nfx** [comment]
i wonder if we can compress it to single command. e.g. `databricks repos checkout /Repos/Production/some-project master`.

**Comment by alexott** [comment]
let's wait for the more APIs then...

**Pull request closed without merging**